In [1]:
import pandas as pd
import numpy as np

Se encuentran datos aqui http://mineducacion.gov.co/sistemasdeinformacion/1735/w3-article-212400.html sobre Inscritos, admitidos, matriculados...

In [2]:
Matriculados_Prime = pd.read_csv("../DatosSPADIES/Matriculados.csv", delimiter=";")

In [29]:
#descriptor para ver por encima las dimensiones del fichero
Matriculados_Prime.describe()

Codigo_Institucion     IES_PADRE  ID_Sector_IES   ID_Caracter  \
count        60363.000000  60363.000000   60363.000000  60363.000000   
mean          3163.344234   3155.605470       1.492785      3.340739   
std           2574.016341   2567.773203       0.499952      0.935865   
min           1101.000000   1101.000000       1.000000      1.000000   
25%           1707.000000   1706.000000       1.000000      3.000000   
50%           2102.000000   2102.000000       1.000000      4.000000   
75%           3201.000000   3201.000000       2.000000      4.000000   
max           9915.000000   9915.000000       2.000000      4.000000   

       Codigo_departamento(IES)  Codigo_Municipio(IES)  Codigo_SNIES_programa  \
count              60363.000000           60363.000000           60363.000000   
mean                  21.259049           21287.513626           53111.167089   
std                   22.336506           22360.396029           41169.257966   
min                    5.000000            5001.000000               1.000000   
25%                   11.000000           11001.000000            9294.000000   
50%                   11.000000           11001.000000           53246.000000   
75%                   17.000000           17001.000000           91410.000000   
max                   91.000000           91001.000000          105995.000000   

       ID_Nivel_Academico  ID_Nivel_Formacion  ID_Metodologia       ID_Area  \
count        60363.000000        60363.000000    60363.000000  60363.000000   
mean             1.289035            4.545400        1.371188      5.691467   
std              0.453318            2.058383        0.631490      2.026603   
min              1.000000            1.000000        1.000000      1.000000   
25%              1.000000            3.000000        1.000000      4.000000   
50%              1.000000            5.000000        1.000000      6.000000   
75%              2.000000            6.000000        2.000000      8.000000   
max              2.000000           10.000000        3.000000      9.000000   

       Codigo_Departamento(Programa)  Codigo_Municipio(Programa)  \
count                   60363.000000                60363.000000   
mean                       32.346984                32463.603035   
std                        27.198638                27224.357865   
min                         5.000000                 5001.000000   
25%                        11.000000                11001.000000   
50%                        18.000000                18001.000000   
75%                        54.000000                54874.000000   
max                        99.000000                99001.000000   

            ID_Sexo     Anio      Semestre  Matriculados_2016  
count  60363.000000  60363.0  60363.000000       60363.000000  
mean       1.495403   2016.0      1.493150          76.427977  
std        0.499983      0.0      0.499957         233.995089  
min        1.000000   2016.0      1.000000           1.000000  
25%        1.000000   2016.0      1.000000           5.000000  
50%        1.000000   2016.0      1.000000          19.000000  
75%        2.000000   2016.0      2.000000          68.000000  
max        2.000000   2016.0      2.000000       16822.000000

In [17]:
Matriculados_Prime.columns

Index(['Codigo_Institucion', 'IES_PADRE',
       'Institucion_Educacion_Superior(IES)', 'Principal_o_Seccional',
       'ID_Sector_IES', 'Sector_IES', 'ID_Caracter', 'Caracter_IES',
       'Codigo_departamento(IES)', 'Departamento_domicilio_IES',
       'Codigo_Municipio(IES)', 'Municipio_domicilio_IES',
       'Codigo_SNIES_programa', 'Programa_Academico', 'ID_Nivel_Academico',
       'Nivel_Academico', 'ID_Nivel_Formacion', 'Nivel_Formacion',
       'ID_Metodologia', 'Metodologia', 'ID_Area', 'Area_Conocimiento',
       'Id_Nucleo', 'Nucleo_Basico_Conocimiento(NBC)',
       'Codigo_Departamento(Programa)', 'Departamento_oferta_del_programa',
       'Codigo_Municipio(Programa)', 'Municipio_oferta_del_programa',
       'ID_Sexo', 'Sexo', 'Anio', 'Semestre', 'Matriculados_2016'],
      dtype='object')

In [31]:
set(Matriculados_Prime["Nivel_Formacion"])

{'Doctorado',
 'Especialización Médico Quirúrgica',
 'Especialización Tecnológica',
 'Especialización Técnico Profesional',
 'Especialización Universitaria',
 'Formación técnica profesional',
 'Maestría',
 'Tecnológica',
 'Universitaria'}

In [45]:
#un head para ver ejemplos de los datos 
Matriculados_Prime.head(10)

Codigo_Institucion  IES_PADRE Institucion_Educacion_Superior(IES)  \
0                1101       1101    UNIVERSIDAD NACIONAL DE COLOMBIA   
1                1101       1101    UNIVERSIDAD NACIONAL DE COLOMBIA   
2                1101       1101    UNIVERSIDAD NACIONAL DE COLOMBIA   
3                1101       1101    UNIVERSIDAD NACIONAL DE COLOMBIA   
4                1101       1101    UNIVERSIDAD NACIONAL DE COLOMBIA   
5                1101       1101    UNIVERSIDAD NACIONAL DE COLOMBIA   
6                1101       1101    UNIVERSIDAD NACIONAL DE COLOMBIA   
7                1101       1101    UNIVERSIDAD NACIONAL DE COLOMBIA   
8                1101       1101    UNIVERSIDAD NACIONAL DE COLOMBIA   
9                1101       1101    UNIVERSIDAD NACIONAL DE COLOMBIA   

  Principal_o_Seccional  ID_Sector_IES Sector_IES  ID_Caracter Caracter_IES  \
0             PRINCIPAL              1    OFICIAL            4  Universidad   
1             PRINCIPAL              1    OFICIAL            4  Universidad   
2             PRINCIPAL              1    OFICIAL            4  Universidad   
3             PRINCIPAL              1    OFICIAL            4  Universidad   
4             PRINCIPAL              1    OFICIAL            4  Universidad   
5             PRINCIPAL              1    OFICIAL            4  Universidad   
6             PRINCIPAL              1    OFICIAL            4  Universidad   
7             PRINCIPAL              1    OFICIAL            4  Universidad   
8             PRINCIPAL              1    OFICIAL            4  Universidad   
9             PRINCIPAL              1    OFICIAL            4  Universidad   

   Codigo_departamento(IES) Departamento_domicilio_IES        ...         \
0                        11                 BOGOTA D.C        ...          
1                        11                 BOGOTA D.C        ...          
2                        11                 BOGOTA D.C        ...          
3                        11                 BOGOTA D.C        ...          
4                        11                 BOGOTA D.C        ...          
5                        11                 BOGOTA D.C        ...          
6                        11                 BOGOTA D.C        ...          
7                        11                 BOGOTA D.C        ...          
8                        11                 BOGOTA D.C        ...          
9                        11                 BOGOTA D.C        ...          

            Nucleo_Basico_Conocimiento(NBC) Codigo_Departamento(Programa)  \
0  Ingeniería agronómica, pecuaria y afines                            11   
1  Ingeniería agronómica, pecuaria y afines                            11   
2  Ingeniería agronómica, pecuaria y afines                            11   
3  Ingeniería agronómica, pecuaria y afines                            11   
4                      Medicina veterinaria                            11   
5                      Medicina veterinaria                            11   
6                      Medicina veterinaria                            11   
7                      Medicina veterinaria                            11   
8                                 Zootecnia                            11   
9                                 Zootecnia                            11   

   Departamento_oferta_del_programa Codigo_Municipio(Programa)  \
0                        BOGOTA D.C                      11001   
1                        BOGOTA D.C                      11001   
2                        BOGOTA D.C                      11001   
3                        BOGOTA D.C                      11001   
4                        BOGOTA D.C                      11001   
5                        BOGOTA D.C                      11001   
6                        BOGOTA D.C                      11001   
7                        BOGOTA D.C                      11001   
8                        BOGOTA D.C                      1100

## <span style="color: blue"> Total de Matriculados en el 2016 de todas las sedes de todas las universidades </span>

In [15]:
sum(Matriculados_Prime["Matriculados_2016"])

4613422

Se puede agrupar los datos por institucion asi:

In [3]:
Matriculados_groups = Matriculados_Prime.groupby('Codigo_Institucion')

In [4]:
Matriculados_groups.groups.keys()

dict_keys([2847, 2830, 1707, 4101, 4102, 1722, 3103, 1708, 4106, 4107, 4108, 4109, 4110, 2733, 4112, 9902, 9906, 9903, 3102, 2848, 2831, 2736, 3107, 1723, 4726, 2737, 3114, 3115, 3116, 2738, 4701, 4709, 1715, 4702, 2102, 2825, 2104, 2740, 2106, 2849, 2110, 2741, 2114, 2827, 1718, 4111, 2828, 1207, 3719, 1101, 1102, 1103, 1104, 2744, 1106, 1107, 1108, 1109, 1110, 1111, 1112, 1113, 1114, 1115, 1117, 1118, 1119, 1120, 1121, 1122, 1123, 4708, 1125, 1126, 2833, 1212, 4714, 2834, 2749, 4721, 1214, 3702, 3703, 2836, 3706, 1215, 3710, 1813, 3712, 1728, 3715, 3204, 2838, 3718, 1729, 3720, 5801, 3724, 1730, 2702, 9910, 2704, 2840, 2707, 2708, 2709, 2710, 2711, 2712, 2713, 2715, 2842, 2206, 1733, 2720, 2209, 2211, 2724, 1701, 1702, 1703, 1704, 1705, 2730, 2731, 2732, 1709, 1710, 1711, 1712, 1713, 1202, 1203, 1204, 1717, 1206, 1719, 1208, 2745, 2746, 2747, 1724, 1725, 1726, 1727, 1216, 1217, 1218, 1219, 1220, 1221, 1734, 1223, 4808, 4810, 4727, 4812, 2850, 4813, 4817, 4818, 2837, 4822, 3705, 3801,

In [8]:
print("Hay:", len(Matriculados_groups.groups.keys()), "grupos (sedes de Educacion Superior)")

Hay: 320 grupos (sedes de Educacion Superior)


Se puede utilizar la siguiente funcion para obtener los grupos segun la institucion y ademas de eso los parametros que nos interezan y la cantidad de matriculados total

In [24]:
def get_enrolled(df):
    return [set(df["Institucion_Educacion_Superior(IES)"]), set("Principal_o_Seccional"), set(df["ID_Sector_IES"]), set(df["Sector_IES"]), set(df["ID_Caracter"]), set(df["Caracter_IES"]), set(df["Codigo_departamento(IES)"]), set(df["Departamento_domicilio_IES"]), set(df["Codigo_Municipio(IES)"]), set(df["Municipio_domicilio_IES"]), set(df["ID_Sexo"]), set(df["Sexo"]), set(df["Semestre"]), np.sum(df["Matriculados_2016"])]

Se puede agrupar por el nombre de la universidad, de esta manera no importa si es la sede principal o no, se obtiene un grupo por cada universidad incluyendo todas las sedes.  
De esta manera podemos reducir el numero de grupos de 320 a 274

In [12]:
Matriculados_groups_info = Matriculados_Prime.groupby('Institucion_Educacion_Superior(IES)')

In [14]:
len(Matriculados_groups_info.groups.keys())

274

In [25]:
tmp = Matriculados_groups_info.apply(get_enrolled)

In [30]:
tmp[0]

[{'CENTRO DE EDUCACION MILITAR - CEMIL'},
 {'P', 'S', '_', 'a', 'c', 'e', 'i', 'l', 'n', 'o', 'p', 'r'},
 {1},
 {'OFICIAL'},
 {3},
 {'Institución Universitaria/Escuela Tecnológica'},
 {11},
 {'BOGOTA D.C'},
 {11001},
 {'SANTAFE DE BOGOTA'},
 {1, 2},
 {'HOMBRE', 'MUJER'},
 {1, 2},
 1341]